astropy/modeling/functional_models.py

```diff
@@ -400,13 +400,15 @@ def evaluate(x, factor):
         return factor * x
 
 
-class Redshift(Fittable1DModel):
+class RedshiftWavelength(Fittable1DModel):
     """
     One dimensional redshift model.
 
+    Redshifts a wavelength array from rest (z=0) to the input redshift.
+
     Parameters
     ----------
-    z : float or a list of floats
+    z : int or float
```

Comment:
> When using a list of z, I get a ValueError: "Model input argument u'x' of shape (XX,) cannot be broadcast with parameter 'z' of shape (YY,)."

If the input and the parameters are both arrays then when you evaluate the model they have to broadcast.  This is just the same as trying to multiply two Numpy arrays of different lengths.